ding/policy/meta_diffuser.py

```diff
@@ -0,0 +1,446 @@
+            # (float) discount factor for the discounted sum of rewards, aka. gamma.
+            discount_factor=0.99,
+            gradient_accumulate_every=2,
+            # train_epoch = train_epoch * gradient_accumulate_every
+            train_epoch=60000,
+
+            # step start update target model and frequence
+            step_start_update_target=2000,
+            update_target_freq=10,
+            # update weight of target net
+            target_weight=0.995,
+            value_step=2e3,
+
+            # dataset weight include returns
+            include_returns=True,
+
+            # (float) Weight uniform initialization range in the last output layer
+            init_w=3e-3,
+        ),
+    )
+
+    def default_model(self) -> Tuple[str, List[str]]:
+        return 'metadiffuser', ['ding.model.template.diffusion']
+    
+    def _init_learn(self) -> None:
+        r"""
+        Overview:
+            Learn mode init method. Called by ``self.__init__``.
+            Init q, value and policy's optimizers, algorithm config, main and target models.
+        """
+        # Init
+        self._priority = self._cfg.priority
+        self._priority_IS_weight = self._cfg.priority_IS_weight
+        self.action_dim = self._cfg.model.diffuser_model_cfg.action_dim
+        self.obs_dim = self._cfg.model.diffuser_model_cfg.obs_dim
+        self.n_timesteps = self._cfg.model.diffuser_model_cfg.n_timesteps
+        self.gradient_accumulate_every = self._cfg.learn.gradient_accumulate_every
+        self.gradient_steps = 1
+        self.update_target_freq = self._cfg.learn.update_target_freq
+        self.step_start_update_target = self._cfg.learn.step_start_update_target
+        self.target_weight = self._cfg.learn.target_weight
+        self.value_step = self._cfg.learn.value_step
+        self.horizon = self._cfg.model.diffuser_model_cfg.horizon
+        self.include_returns = self._cfg.learn.include_returns
+        self.eval_batch_size = self._cfg.learn.eval_batch_size
+        self.warm_batch_size = self._cfg.learn.warm_batch_size
+        self.test_num = self._cfg.learn.test_num
+        self.have_train = False
+        self._forward_learn_cnt = 0
+        self.encoder_len = self._cfg.learn.encoder_len
+
+        self._plan_optimizer = Adam(
+            self._model.diffuser.model.parameters(),
+            lr=self._cfg.learn.learning_rate,
+        )
+
+        self._pre_train_optimizer = Adam(
+            list(self._model.reward_model.model.parameters()) + list(self._model.embed.parameters()) \
+                  + list(self._model.dynamic_model.parameters()),
+            lr=self._cfg.learn.learning_rate,
+        )
+
+        self._gamma = self._cfg.learn.discount_factor
+
+        self._target_model = copy.deepcopy(self._model)
+
+        self._learn_model = model_wrap(self._model, wrapper_name='base')
+        self._learn_model.reset()
+
+    def _forward_learn(self, data: List[torch.Tensor]) -> Dict[str, Any]:
```

Comment:
data should be collated into batchsize before entering policy._forward_learn.
data type shoule be Dict[str, torch.Tensor].